atarashi/agents/semanticSearch.py

```diff
@@ -0,0 +1,208 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Copyright 2024 Abdelrahman Jamal (abdelrahmanjamal5565@gmail.com)
+
+SPDX-License-Identifier: GPL-2.0
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+version 2 as published by the Free Software Foundation.
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+"""
+import argparse
+import os
+import nirjas
+import numpy as np
+import pandas as pd
+from fuzzywuzzy import fuzz
+
+
+from atarashi.agents.atarashiAgent import AtarashiAgent, exactMatcher
+
+__author__ = "Abdelrahman Jamal"
+__email__ = "abdelrahmanjamal5565@gmail.com"
+
+class SemanticSearchAgent(AtarashiAgent):
+    """
+    An agent that performs semantic search to identify potential licenses within files.
+    """
+
+    def __init__(self, licenseList):
+        super().__init__(licenseList)
```

Comment:
```suggestion
    def __init__(self, licenseList, verbose=0):
        super().__init__(licenseList)
```

If verbose type output is planned, The input flag for verbose is defined but not passed. Prone to throw error